spec/controllers/new_gestionnaire/procedures_controller_spec.rb

```diff
@@ -336,4 +336,25 @@
       end
     end
   end
+
+  describe '#update_email_notifications' do
+    let(:gestionnaire) { create(:gestionnaire) }
+    let!(:procedure) { create(:procedure, gestionnaires: [gestionnaire]) }
+
+    context "when logged in" do
+      before { sign_in(gestionnaire) }
+
+      it { expect(procedure.gestionnaires_with_email_notifications).to match([]) }
+
+      context 'when the gestionnaire update its preferences' do
+        let(:assign_to) { gestionnaire.assign_to.find_by(procedure: procedure) }
+
+        before do
+          patch :update_email_notifications, params: { procedure_id: procedure.id, assign_to: { id: assign_to.id, email_notifications: true } }
+        end
+
+        it { expect(procedure.gestionnaires_with_email_notifications).to match([gestionnaire]) }
```

Comment:
Pareil, je préférerais `expect(...).to eq([gestionnaire])` (ou `expect(...).to contain_exactly(gestionnaire)`)